src/runtime/hashmap.h

```diff
@@ -0,0 +1,405 @@
+#ifndef HALIDE_RUNTIME_HASHMAP_H
+#define HALIDE_RUNTIME_HASHMAP_H
+
+// the bulk of this hashmap implementation is based on 'cache.cpp'
+
+#include "printer.h"
+#include "scoped_mutex_lock.h"
+
+namespace Halide { namespace Runtime { namespace Internal {
+
+WEAK bool keys_equal(const uint8_t *key1, const uint8_t *key2, size_t key_size) {
+    return memcmp(key1, key2, key_size) == 0;
+}
+
+WEAK uint32_t djb_hash(const uint8_t *key, size_t key_size) {
```

Comment:
Is it possible to have this inlined?  Or is the intent a user may want to override the hash function?

Reply:
Same thing here: I'd like to hear @zvookin 's opinion.

Reply:
Yeah, it can be always inline. Hopefully changing it won't result in COMDAT errors on the Mac.

Reply:
Absolutely should use always-inline here, regardless of COMDAT errors. If this was intended to be user-overridable it would be documented and the name would be prefixed with "halide_".